### PR TITLE
Prepare Cortext for open source

### PR DESCRIPTION
**Pull request by priethor**
## What

Prepares Cortext for public source release by adding clear prototype warnings, contribution and security guidance, GPL licensing notes, and simpler public docs focused on Cortext as a WordPress knowledge base.

## Why

The repo should be clear about what is safe to expect today: Cortext is experimental, early data is disposable, and the public docs should avoid over-promising.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+## Branches
+
+Use a short branch prefix followed by a concise slug:
+
+-   `add/`
+-   `fix/`
+-   `docs/`
+-   `refactor/`
+-   `tests/`
```
Comment (mcsf) [inline CONTRIBUTING.md]: Nit: some of the example prefixes are action-based, others thematic. Is it intentional? Personally, I don't care all that much about strict naming of branches, as long as the author makes an earnest attempt to give the branch a self-explanatory name.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+```sh
+npm run format:check
+npm run lint:js
+npm run lint:style
+composer run phpcs
+npm run test:unit
+composer run test:php
+```
+
+For UI or routing changes, run the relevant Playwright coverage with `npm run test:e2e` too.
```
Comment (mcsf) [inline CONTRIBUTING.md]: Have we addressed the dev-vs-test environment matter? Which env does `test:e2e` currently assume? Shouldn't readers be told in this doc about wp-env?

File: cortext.php [hunk 3]
```diff
@@ -2,7 +2,7 @@
 /**
  * Plugin Name:       Cortext
  * Plugin URI:        https://github.com/priethor/cortext
- * Description:       Notion-inspired workspace inside WordPress: nested pages, typed collections with multiple views, and cross-type taxonomies that add fields to individual rows.
+ * Description:       Knowledge base inside WordPress: nested pages, typed collections with multiple views, and cross-type taxonomies that add fields to individual rows.
```
Comment (mcsf) [inline cortext.php]: Feels like we need more pep here, but I'm feeling uninspired at the moment.

File: README.md [hunk 4]
```diff
@@ -2,36 +2,48 @@
 
 # Cortext
```
Comment (mcsf) [inline README.md:3]: Similarly to the plugin description, I think we may come back to this file soon to make it more immediately apparent what it's all about, why one may want it, etc. But for the purpose of this PR, this is fine IMO. :)

File: docs/licensing.md [hunk 5]
```diff
@@ -0,0 +1,9 @@
+# Licensing
```
Comment (mcsf) [inline docs/licensing.md]: This whole file seems like a mouthful, no? Projects often have a license/LICENSE document with the actual license, or may just point to an external reference. I would understand explaining the point of the license if we had made an unusual choice, but going with GPL2 is just the obvious thing. Less is more, IMO.

File: docs/getting-started.md [hunk 6]
```diff
@@ -4,11 +4,11 @@ Cortext is an early prototype. There is no packaged release yet; to try it you r
...
 
-- PHP 8.1+
-- WordPress 6.9+ (provided by wp-env)
-- Docker (recommended runtime for wp-env)
-- Node.js 20+ (matching `@wordpress/scripts`)
-- Git
+-   PHP 8.1+
+-   WordPress 6.9+ (provided by wp-env)
+-   Docker (recommended runtime for wp-env)
+-   Node.js 24.15 (matching `package.json` and CI)
```
Comment (mcsf) [inline docs/getting-started.md]:
```suggestion
-   Node.js 24.15
```

File: docs/roadmap.md [hunk 7]
```diff
@@ -1,28 +1,31 @@
...
+## Current prototype
 
-Baseline goals for the month:
+The repo already has a usable first shape:
 
-- A full-screen admin shell with sidebar navigation, nested pages, and collection views.
-- In-canvas content modeling and editing powered by DataForm and DataViews.
-- Cross-type taxonomies (catchy name TBD) that add fields to individual rows.
-- A v0 Notion importer.
+-   A full-screen admin shell.
```
Comment (mcsf) [inline docs/roadmap.md:9]: Super super nit: this is a weird spacing choice for list items :P 

(You don't need to change anything because of my comment, but I still wanted to obsess over this!)

File: docs/roadmap.md [hunk 8]
```diff
@@ -1,28 +1,31 @@
...
+-   Nested pages backed by Gutenberg.
```
Comment (mcsf) [inline docs/roadmap.md]: What do we mean by "backed by Gutenberg"? I presume "Gutenberg" to be a sort of synedocque for "WordPress", but — even then — what would we mean by "backed by WordPress"?

File: docs/roadmap.md [hunk 9]
```diff
@@ -1,28 +1,31 @@
...
+-   Typed collections with table, grid, and list views.
+-   Inline row editing, relation fields, rollups, and row details.
```
Comment (mcsf) [inline docs/roadmap.md]: I'd probably make this two separate items:

```suggestion
-   Inline editing of rows
-   Relation fields, _rollups_, and row details.
```

or

```suggestion
-   Inline editing of fields
-   Relation fields, _rollups_, and row details.
```

File: docs/roadmap.md [hunk 10]
```diff
@@ -1,28 +1,31 @@
...
+-   Collection views that can be embedded inside pages.
+-   Basic public rendering for Cortext pages.
```
Comment (mcsf) [inline docs/roadmap.md]: Sounds opaque, IMO. It could be more along the lines of "Basic publishing abilities for Cortext pages."

File: docs/roadmap.md [hunk 11]
```diff
@@ -1,28 +1,31 @@
...
+-   Experimental shell theming.
```
Comment (mcsf) [inline docs/roadmap.md]: Does "shell" apply to both editor and frontend, or just the former? "Theming" is a recognisable term in the context of user applications, but "shell" is IMO more subject to interpretation, hence my question.

File: docs/roadmap.md [hunk 12]
```diff
@@ -1,28 +1,31 @@
...
 
-Stretch goals:
+## Before this is production-ready
 
-- Basic `Formula` type.
-- Relationships between collections.
-- Frontend rendering.
+The storage model, REST responses, block attributes, and theme tokens still need
+to settle. Until then, early data is disposable and migrations are not planned.
```
Comment (mcsf) [inline docs/roadmap.md]: To make as clear as possible, how 'bout:

```suggestion
The storage model, REST responses, block attributes, and theme tokens still
need to settle. **Until then, data schemas are volatile, user data is assumed
disposable, and migrations are not planned.**
```

File: docs/roadmap.md [hunk 13]
```diff
@@ -1,28 +1,31 @@
...
+The next work should focus on:
 
-- Social media feedback and interactions.
-- Plugin active installs on WordPress.org.
-- **Bonus**: issues or PRs filed against Gutenberg or WP Core based on gaps found during the work.
+-   Tightening the core page and collection workflows.
+-   Hardening permissions, tests, accessibility, and editor edge cases.
+-   Deciding which APIs are internal and which can become public.
+-   Designing import/export around multiple sources, not one source too early.
+-   Adding migrations only after the data model slows down.
```
Comment (mcsf) [inline docs/roadmap.md]:
```suggestion
-   Adding migrations only after the data model stabilizes somewhat.
```

File: docs/roadmap.md (hunk 12 above)
Comment (mcsf) [reply docs/roadmap.md]: Oh, I just noticed that this text was formatted to around 80 chars. Me, I'm an old soul, so this was second-nature to me. But then I realised that, prior to this PR, Markdown text was left unformatted… which matches the general trend nowadays (and which, even though I dislike it, is more Git-friendly).

Anyway, my questions are:

- Should we settle on a single convention?
- Was your formatting choice intentional? (And what did you use? For me, with Vim, it's [muscle memory](https://vimhelp.org/change.txt.html#formatting).)

File: docs/vision.md [hunk 14]
```diff
@@ -2,34 +2,39 @@
 
 ## The problem
 
-Notion is great at structured knowledge work. Databases, views, relations, inline databases inside pages, super tags. But your data lives on someone else's servers, policies can change, and the content model isn't something you can extend.
+Knowledge bases need documents, typed records, views, and relations. WordPress
+has the storage pieces for that work, but the editing experience is scattered
+across custom fields, list tables, plugin screens, and theme templates.
 
-WordPress owns the data story. Self-hosted, exportable, stable for over twenty years. What WordPress has never offered is the Notion experience. The block editor handles documents well. Everything structured (custom fields, relations, views) has historically lived in ACF/SCF territory, and the stitching has always felt a step behind what Notion does natively.
+WordPress owns the data story. It is self-hosted, exportable, and stable. The block editor handles documents well. What is missing is a knowledge base interface that treats pages, collections, fields, and views as one product surface while keeping the data in WordPress.
```
Comment (mcsf) [inline docs/vision.md]: I don't like and don't understand "WordPress owns the data story.", but the rest of the paragraph is good.

File: docs/vision.md [hunk 15]
```diff
@@ -2,34 +2,39 @@
...
 
-WordPress core recently shipped three primitives that change the calculus:
-
-1. `@wordpress/dataviews` (6.7+): a filterable, sortable, multi-layout list/grid/table component. Source-agnostic, mountable anywhere.
-2. `@wordpress/editor` plus `BlockCanvas`: a standalone block editor you can mount outside the usual post-editor chrome.
-3. Block Bindings (6.5+): any block attribute can source its value from an external provider.
-
-With these in place, the gap between WordPress and a Notion-inspired UX is no longer a missing primitive. It's a missing composition. Cortext is that composition.
+WordPress core now has enough of the pieces to make this practical: posts, post
+meta, REST, Gutenberg, and DataViews. Cortext wires those together into one
```
Comment (mcsf) [inline docs/vision.md]: What is "Gutenberg" here? :)

File: docs/vision.md [hunk 16]
```diff
@@ -2,34 +2,39 @@
...
+Every architectural decision should start with WordPress primitives. No custom
+tables unless the post/meta model clearly fails. No REST API bypass. No fighting
+the block editor.
 
-If something is hard to do the WordPress way, that's a signal to reconsider, not to route around. Sometimes the reconsideration reveals a misuse of the primitive; other times it reveals a genuine gap in core, and the right answer is to fix it upstream. Because Cortext is a prototype, forking an experimental component locally to iterate on it quickly is acceptable; the goal in that case is to land the improvement upstream once it proves out. A parallel implementation that outlives the iteration cycle is technical debt.
+If something is hard to do the WordPress way, that is a signal to reconsider the shape of the feature. Sometimes we are using the primitive wrong. Sometimes core has a real gap, and the better answer is to fix it upstream. Cortext is a prototype, so a short-lived local fork of an experimental component is fine while we learn. A parallel implementation that sticks around after the experiment is technical debt.
```
Comment (mcsf) [inline docs/vision.md:19]: See my comment somewhere else about text formatting (line wrapping). This paragraph is unwrapped, while the rest of the file — except for the very last paragraph — is wrapped.

File: docs/vision.md [hunk 17]
```diff
@@ -2,34 +2,39 @@
...
+Cortext is WordPress under the hood, but the workspace UI is built for maintaining a knowledge base rather than doing ordinary wp-admin chores. wp-admin stays available for operators who want it; everyone else works in a dedicated shell.
 
-The shell is themeable, but the API is deliberately small. Operators can retune Cortext chrome with tokens. They cannot move the sidebar, reshape DataViews, or replace the inspector. Content pages keep the full WordPress block-theme API, so the workspace can feel like Cortext while published pages still feel like the site. See [Theming](theming.md) for the contract and roadmap.
+The shell is themeable, but the API is intentionally small. Operators can retune Cortext chrome with tokens. They cannot move the sidebar, reshape DataViews, or replace the inspector. Content pages keep the full WordPress block-theme API, so the workspace can feel like Cortext while published pages still feel like the site. See [Theming](theming.md) for the current contract notes.
```
Comment (mcsf) [inline docs/vision.md]: What is this one:

> Operators can retune Cortext chrome with tokens

File: docs/theming.md [hunk 18]
```diff
@@ -1,92 +1,30 @@
...
-Keep two surfaces apart.
+These are implementation notes for the current prototype, not a stable public
+API.
 
-The shell is Cortext's UI: sidebar, toolbar buttons, empty states, and the frame around the work area. Cortext owns those pieces. Shell themes change them through CSS custom properties on `.cortext-root`.
+Cortext has two visual surfaces:
 
-Content is the other side. Published pages and the inside of Gutenberg's editor iframe follow the active WordPress block theme. Cortext does not add shell tokens to `wp_head` or pass shell colors into the iframe.
+-   The shell: sidebar, toolbar, buttons, empty states, and collection views.
+-   Content: published pages and the Gutenberg editor canvas.
```
Comment (mcsf) [inline docs/theming.md]: "block editor canvas"? something like that?

File: docs/architecture.md [hunk 19]
```diff
@@ -1,105 +1,50 @@
...
+## Storage
 
--   we create a new database
+Cortext currently uses WordPress posts and post meta for the main data model:
 
-```php
-$collection_id = wp_insert_post( 'crtxt_collection', [ ... ] );
-register_post_type( 'crtxt_books', [ ... ] );
-```
+-   Pages are hierarchical WordPress posts with Gutenberg content.
```
Comment (mcsf) [inline docs/architecture.md]: Here again a mention of Gutenberg

File: docs/architecture.md [hunk 20]
```diff
@@ -1,105 +1,50 @@
...
+allowed to change. Do not build external integrations against them yet.
 
-```php
-$book_id = wp_insert_post( 'crtxt_books', $data );
-```
+## Admin shell
 
--   we add a new column
+The main product surface is a full-screen React app in wp-admin. It has a
+sidebar for pages and collections, a Gutenberg canvas for documents, and
```
Comment (mcsf) [inline docs/architecture.md]: and here

File: docs/architecture/shell.md [hunk 21]
```diff
@@ -1,40 +1,36 @@
 # Shell architecture
 
-The Cortext shell is a React SPA hosted on a full-screen wp-admin page. Running under `is_admin()` means REST nonces, user capabilities, and block asset defaults (`wp_should_load_separate_core_block_assets()` is false) are already in place, so the editor can be mounted without bespoke plumbing.
+Cortext runs as a React app inside wp-admin. The shell is full-screen, but it
+still uses WordPress authentication, permissions, REST nonces, editor settings,
+and block assets.
```
Comment (mcsf) [inline docs/architecture/shell.md]: Might be good to add something about `core-data` and/or related subsystems.

File: docs/modeling-guide.md [hunk 22]
```diff
@@ -1,88 +1,47 @@
...
 
-The plain-English rule:
+## When to add a collection
 
-> A row's fields = its collection's fields, plus the fields of every cross-type tag stuck on it.
+Add a collection when the rows share the same basic shape.
 
-Two rows in the same collection can have different fields if they carry different cross-type tags.
+For example, a Books collection might have fields for author, status, rating, and
+date finished. A People collection might have fields for email, company, and
```
Comment (mcsf) [inline docs/modeling-guide.md]: "status" is a little surprising. Maybe:


```suggestion
For example, a Books collection might have fields for author, ISBN, rating, and
date finished. A People collection might have fields for email, company, and
```

File: docs/modeling-guide.md [hunk 23]
```diff
@@ -1,88 +1,47 @@
...
+role. Those are different enough to deserve separate collections.
 
-## Worked example
+Do not add a collection just because a few rows need an extra field. That is how
+workspaces turn into a maze of near-duplicates.
```
Comment (mcsf) [inline docs/modeling-guide.md]: What should people do in those scenarios, then? Add the fields, even though few rows will use them?

File: docs/modeling-guide.md [hunk 24]
```diff
@@ -1,88 +1,47 @@
...
 
-You keep all your notes in a single Notes collection, with fields: `title` (text), `created_at` (datetime).
+## When a view is enough
```
Comment (mcsf) [inline docs/modeling-guide.md]: At the top of the document, we explain what a collection is, but never mention views. It's awkward to have a section title go straight to "when a ~ is enough".

Perhaps the section could be preceded by a paragraph acting as a bridge from collections to views. Roughly:

```diff
 Do not add a collection just because a few rows need an extra field. That is
-how workspaces turn into a maze of near-duplicates.
+how workspaces turn into a maze of near-duplicates. In addition, "views" can
+come in handy when there is more than one way for a given collection of data
+to be understood, filtered or organized.

## When a view is enough
```

Comment (mcsf) [reply docs/modeling-guide.md]: Eventually we may even add screenshots to illustrate both concepts. I know that at the moment the UI is fast-evolving, so it may feel pointless.

**Review by mcsf — COMMENTED** [review]
Looks pretty good! Thanks for all the work you've been doing.

Aside from all the comments I've left, some high-level things:

- Let's clarify our intentions around the use of the name "Gutenberg". My default would be to avoid it and be clear in each use, e.g. "block-based editor" (or just "block editor"), or simply "WordPress", etc.
- Let's settle on formatting conventions for Markdown files. It might be easier to just go with no line wrapping (one paragraph <-> one line), but I'm open to alternatives.
- `docs/architecture/data-model-intermediate-json.md`, though not touched by this PR, still references a specific product. Given the effort taken to weed out mentions throughout the rest of the codebase, shouldn't we remove them there too, at least for now?

File: CONTRIBUTING.md (hunk 1 above)
Comment (priethor) [reply CONTRIBUTING.md]: IIRC, in Gutenberg, we also combine actions with themes. I'm relaxing the requirement from specific prefixes to just being self-explanatory.

File: README.md (hunk 4 above)
Comment (priethor) [reply README.md:3]: Agreed!

File: docs/roadmap.md (hunk 11 above)
Comment (priethor) [reply docs/roadmap.md]: Shell only refers to the admin workspace; public theming is the classic WP one, so this needs reframing to make it clear that we are referring about the workspace editor.

File: docs/vision.md (hunk 14 above)
Comment (priethor) [reply docs/vision.md]: Thanks; yep, AI slop leftover. Removing it.

File: docs/roadmap.md (hunk 12 above)
Comment (priethor) [reply docs/roadmap.md]: Agreed on the stronger warning and the bold emphasis. I also think we should follow up with a first-run warning modal so users see this before creating data 😅

On wrapping: there is no strict convention in the repo right now, and the wrapping in this PR was not deliberate. I do not have a strong opinion, though _I think_ lean toward one paragraph per line to avoid noisy diffs. Let's handle Markdown formatting as a separate standardization follow-up.

File: docs/vision.md (hunk 17 above)
Comment (priethor) [reply docs/vision.md]: Implementation details, fixed i!

File: docs/modeling-guide.md (hunk 22 above)
Comment (priethor) [reply docs/modeling-guide.md]: Haha, I see what you mean! Let's change it to ISBN, yes.

For context about status, it's quite common to track books/videogame libraries in these kinds of apps and add a status column to track whether you've gone through it already, etc, as if they were tasks. _Although as I'm writing this, it sounds a lot like converting a hobby into a chore._ 🙂

**Comment by priethor** [comment]
Thanks for the careful review! ❤️ 

> docs/architecture/data-model-intermediate-json.md, though not touched by this PR, still references a specific product. Given the effort taken to weed out mentions throughout the rest of the codebase, shouldn't we remove them there too, at least for now?

I left that file as-is on purpose. The rest of this PR tries to remove product references from public-facing copy, but this file is different: it’s a technical note for one specific import shape. If we add imports from more tools later, I’d expect each one to have its own mapping notes too.

That said, I agree it should be clearer that it is a technical sketch for importer documentation, not product positioning.

> Let's clarify our intentions around the use of the name "Gutenberg". My default would be to avoid it and be clear in each use, e.g. "block-based editor" (or just "block editor"), or simply "WordPress", etc.

100%. That was sloppy on my part; I should have caught it earlier. I’ll clean those up case by case, so we use “block editor”, “WordPress”, or the specific package name where it fits. 🤦

File: docs/roadmap.md (hunk 12 above)
Comment (priethor) [reply docs/roadmap.md]: Normalized formatting to single line per paragraph in db1f630 for consistency, but let's follow up after this PR.

**Comment by priethor** [comment]
> Let's settle on formatting conventions for Markdown files. It might be easier to just go with no line wrapping (one paragraph <-> one line), but I'm open to alternatives.

Normalized formatting to single line per paragraph in db1f630 for consistency, but let's follow up after this PR.

File: docs/roadmap.md (hunk 7 above)
Comment (priethor) [reply docs/roadmap.md:9]: Yeah, it does 🥴 . This is coming from the WordPress/Prettier formatting preset, so I’d rather not hand-format around it in this PR and fight `format:check`.
